placeholder/src/commonMain/kotlin/com/eygraber/compose/placeholder/PlaceholderNode.kt

```diff
@@ -0,0 +1,348 @@
+import androidx.compose.animation.core.AnimationSpec
+import androidx.compose.animation.core.MutableTransitionState
+import androidx.compose.animation.core.RepeatMode
+import androidx.compose.animation.core.spring
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.geometry.toRect
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.Outline
+import androidx.compose.ui.graphics.Paint
+import androidx.compose.ui.graphics.RectangleShape
+import androidx.compose.ui.graphics.Shape
+import androidx.compose.ui.graphics.drawOutline
+import androidx.compose.ui.graphics.drawscope.ContentDrawScope
+import androidx.compose.ui.graphics.drawscope.DrawScope
+import androidx.compose.ui.graphics.drawscope.drawIntoCanvas
+import androidx.compose.ui.node.DrawModifierNode
+import androidx.compose.ui.node.ModifierNodeElement
+import androidx.compose.ui.node.invalidateDraw
+import androidx.compose.ui.platform.InspectorInfo
+import androidx.compose.ui.unit.LayoutDirection
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.yield
+
+public fun Modifier.placeholderNode(
+  visible: Boolean,
+  color: Color,
+  shape: Shape = RectangleShape,
+  highlight: PlaceholderHighlight? = null,
+  placeholderFadeAnimationSpec: AnimationSpec<Float> = spring(),
+  contentFadeAnimationSpec: AnimationSpec<Float> = spring() ,
+): Modifier = this then PlaceholderElement(
+  visible = visible,
+  color = color,
+  shape = shape,
+  highlight = highlight,
+  placeholderFadeAnimationSpec = placeholderFadeAnimationSpec,
+  contentFadeAnimationSpec = contentFadeAnimationSpec,
+)
+
+private class PlaceholderElement(
+  private val visible: Boolean,
+  private val color: Color,
+  private val shape: Shape,
+  private val highlight: PlaceholderHighlight?,
+  private val placeholderFadeAnimationSpec: AnimationSpec<Float>,
+  private val contentFadeAnimationSpec: AnimationSpec<Float>,
+) : ModifierNodeElement<PlaceholderNode>() {
+  override fun create(): PlaceholderNode = PlaceholderNode(
+    visible = visible,
+    color = color,
+    shape = shape,
+    highlight = highlight,
+    placeholderFadeAnimationSpec = placeholderFadeAnimationSpec,
+    contentFadeAnimationSpec = contentFadeAnimationSpec
+  )
+
+  override fun update(node: PlaceholderNode) {
+    node.apply {
+      updateVisible(visible)
+      updateColor(color)
+      updateShape(shape)
+      updateHighlight(highlight)
+      updatePlaceholderFadeAnimationSpec(placeholderFadeAnimationSpec)
+      updateContentFadeAnimationSpec(contentFadeAnimationSpec)
+    }
+  }
+
+  override fun InspectorInfo.inspectableProperties() {
+    name = "placeholder"
+    value = visible
+    properties["visible"] = visible
```

Comment:
Removing either of these would cause tests to fail. Not sure if this was intentional, or just an oversight when the accompanist library was written, but I'll leave it for now. Thanks!